contracts/multisig-prover/src/execute.rs

```diff
@@ -317,8 +318,34 @@ pub fn should_update_worker_set(
     cur_workers: &WorkerSet,
     max_diff: usize,
 ) -> bool {
-    new_workers.signers.difference(&cur_workers.signers).count()
-        + cur_workers.signers.difference(&new_workers.signers).count()
+    let new_workers_signers = new_workers
+        .signers
+        .iter()
+        .map(|(address, (weight, pub_key))| Signer {
+            address: Addr::unchecked(address.clone()),
+            weight: *weight,
+            pub_key: pub_key.clone(),
+        })
```

Comment:
Do you need the `map` here? Can you just collect the tuples?

Reply:
In my code I have to convert the `String` address into an `Addr`.
Is this possible without a `map()`?

Reply:
`BTreeSet` already provides a `contains` function, can't you use that directly?

Reply:
even better, have a look at the `BTreeSet` `intersection()` function https://doc.rust-lang.org/std/collections/struct.BTreeSet.html#method.intersection

Reply:
What's wrong with collecting into that datatype? Can't we just do the comparison with that datatype.

@cgorenflo It's not a `BTreeSet` though, it's a `BTreeMap`

Reply:
ah damn, github showed me the version before the change 😠

Reply:
`BTreeMap` is back to using `Signer` instead of tuples as its value, and so this no longer needed.